### PR TITLE
Webpack: Try naming things in webpack 2

### PR DESCRIPTION
**Pull request by blowery**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -235,11 +239,6 @@ if ( calypsoEnv === 'development' ) {
 	}
 } else {
 	webpackConfig.entry.build = path.join( __dirname, 'client', 'boot', 'app' );
-	webpackConfig.plugins = webpackConfig.plugins.concat( [
-		new webpack.HashedModuleIdsPlugin(),
-		new WebpackChunkHash(),
```
Comment (blowery) [inline webpack.config.js]: unsure if we need chunkhash if we also have named modules and hashed chunks. doesn't appear so, but testing is needed.

File: server/bundler/babel/babel-plugin-transform-wpcalypso-async/index.js [hunk 2]
```diff
@@ -98,10 +98,11 @@ module.exports = ( { types: t } ) => {
 					// Transform to asynchronous require.ensure
 					path.replaceWith(
 						t.callExpression( t.memberExpression( t.identifier( 'require' ), t.identifier( 'ensure' ) ), [
-							argument,
+							t.arrayExpression( [ argument ] ),
```
Comment (blowery) [inline server/bundler/babel/babel-plugin-transform-wpcalypso-async/index.js]: these are both moves to align with the new preferred require.ensure calling convention

File: server/bundler/babel/babel-plugin-transform-wpcalypso-async/index.js [hunk 3]
```diff
@@ -98,10 +98,11 @@ module.exports = ( { types: t } ) => {
...
 							t.functionExpression( null, [ t.identifier( 'require' ) ], t.blockStatement( [
 								t.expressionStatement( requireCall )
 							] ) ),
+							t.nullLiteral(),
```
Comment (blowery) [inline server/bundler/babel/babel-plugin-transform-wpcalypso-async/index.js]: should we provide an error handler...?

**Comment by samouri** [comment]
@blowery: lets wait for the perf boost to get merged in next week, then we can follow the guide precisely and also upgrade to webpack3 in the same go!

https://github.com/webpack/enhanced-resolve/pull/92#issuecomment-311861836

**Comment by samouri** [comment]
@blowery: lets continue working on this but bump version to 3.x so that we can use NamedChunkIds.  Then when the perf fix gets into 3 we can bump it one more time, test, and merge!

**Pull request closed without merging**